pkg/platform/kube/client/deployer.go

```diff
@@ -146,6 +159,215 @@ func (d *Deployer) Deploy(ctx context.Context,
 	}, updatedFunctionInstance, "", nil
 }
 
+func (d *Deployer) ScrubFunctionConfig(ctx context.Context,
+	functionConfig *functionconfig.Config) (*functionconfig.Config, error) {
+	var err error
+
+	// get existing function secret
+	functionSecretMap, err := d.getFunctionSecretMap(ctx, functionConfig.Meta.Name, functionConfig.Meta.Namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get function secret")
+	}
+
+	// scrub the function config
+	d.logger.DebugWithCtx(ctx, "Scrubbing function config", "functionName", functionConfig.Meta.Name)
+
+	scrubbedFunctionConfig, secretsMap, err := functionconfig.Scrub(functionConfig,
+		functionSecretMap,
+		d.platform.GetConfig().SensitiveFields.CompileSensitiveFieldsRegex())
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to scrub function config")
+	}
+
+	// scrubbing removes unexported fields, so we need to re-set them
+	scrubbedFunctionConfig.Spec.Resources = functionConfig.Spec.Resources
```

Comment:
That is not maintainable. we should see how gosecretive can take care of it better. we will surely forget to add fields here and this is very error prone

Reply:
Agreed.
I scoured the web for a way to just copy the private fields as is to the scrubbed object, but to no avail...
We will need to find another solution.